File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        hashtable = {}

        for i in strs:
            sortedword = ''.join(sorted(i))
            if sortedword not in hashtable:
                hashtable[sortedword] = []
            hashtable[sortedword].append(i)

        ans = list(hashtable.values())

        return ans

        
        

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for(String str: strs){
            char[] charArray = str.toCharArray();
            Arrays.sort(charArray);
            String signature = new String(charArray);

            if(!map.containsKey(signature)){
            map.put(signature,new ArrayList<>());
            }

            map.get(signature).add(str);
        }

        return new ArrayList<>(map.values());

}
}